### PR TITLE
17 get channel data using shared key

### PR DESCRIPTION
**Pull request by dominic22**

### CODE REVIEW

File: api/src/routes/channel/index.ts [hunk 1]
```diff
@@ -11,14 +11,14 @@ export class ChannelRoutes {
...
-			const { topics, seed, encrypted } = req.body as CreateChannelBody;
+			const { topics, seed, encrypted, hasPresharedKey, presharedKey, subscriptionPassword } = req.body as CreateChannelBody;
 			const { identityId } = req.user;
 
 			if (!identityId) {
 				return res.sendStatus(StatusCodes.BAD_REQUEST);
 			}
 
-			const channel = await this.channelService.create(identityId, topics, encrypted, seed);
+			const channel = await this.channelService.create(identityId, topics, encrypted, hasPresharedKey, seed, presharedKey, subscriptionPassword);
```
Comment (sobolev-alexey) [inline api/src/routes/channel/index.ts:21]: To not accidentially mess up the order of parameters, I would suggest to send them as an object. 
In this case you will need to modify the `create()` function to also accept one object and take parameters from there.
```
const channel = await this.channelService.create({ identityId, topics, encrypted, hasPresharedKey, seed, presharedKey, subscriptionPassword });  // now parameters can be in any order
```

File: api/src/routes/subscription/tests/request-subscription.test.ts [hunk 2]
```diff
@@ -117,7 +119,105 @@ describe('test request subscription route', () => {
 			type: SubscriptionType.Subscriber
 		};
 
-		expect(requestSubscriptionSpy).toHaveBeenCalledWith('testaddress', undefined);
+		expect(requestSubscriptionSpy).toHaveBeenCalledWith('testaddress', seed, presharedKey);
```
Comment (sobolev-alexey) [inline api/src/routes/subscription/tests/request-subscription.test.ts:122]: Just to be sure: you are requesting a subscription with undefined seed and undefined presharedKey, and it still should proceed?

File: api/src/services/channel-service.ts [hunk 3]
```diff
@@ -23,8 +24,21 @@ export class ChannelService {
 		this.password = config.statePassword;
 	}
 
-	create = async (identityId: string, topics: Topic[], encrypted: boolean, seed?: string): Promise<CreateChannelBodyResponse> => {
-		const res = await this.streamsService.create(seed);
+	async create(
```
Comment (sobolev-alexey) [inline api/src/services/channel-service.ts:27]: If you turn the parameters list into an object, the order of parameters won’t matter

File: api/src/routes/subscription/tests/request-subscription.test.ts (hunk 2 above)
Comment (dominic22) [reply api/src/routes/subscription/tests/request-subscription.test.ts:122]: yes in this case the seed and presharedKey will be generated by the api :)

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**